library/Fission/CLI/Command/Watch.hs

```diff
@@ -102,32 +103,14 @@ watcher runner Watch.Options {..} = do
 
       IPFS.addDir ignoredFiles absPath >>= putErrOr \cid@(CID hash) -> do
         UTF8.putText $ "👀 Watching " <> Text.pack absPath <> " for changes...\n"
-
-        sendRequestM (updateApp url cid copyFiles) >>= putErrOr \_ -> do
+        req <- App.update url cid copyFiles
+        retryOnStatus [status502, status504] 100 req >>= putErrOr \_ -> do
           liftIO $ FS.withManager \watchMgr -> do
             hashCache <- newMVar hash
             timeCache <- newMVar =<< getCurrentTime
             void $ handleTreeChanges runner url copyFiles timeCache hashCache watchMgr absPath
             forever . liftIO $ threadDelay 1000000 -- Sleep main thread
 
-updateApp ::
-  ( MonadIO      m
-  , MonadTime    m
-  , MonadLogger  m
-  , ServerDID    m
-  , MonadWebAuth m Token
-  , MonadWebAuth m Ed25519.SecretKey
-  )
-  => URL
-  -> CID
-  -> Bool
-  -> m (ClientM NoContent)
-updateApp url cid copyFiles =
-  authClient (Proxy @App.Update)
```

Comment:
I remember client functions being split out in their own modules. Is there a reason we have them sitting closer to where they're used?

I split this one out again to increase reusability (used in both `Watch` & `Up`)

Reply:
Because it was only being used in the one place. It's specific to this use case. But yeah, sure own module is fine 👍